pkg/reconciler/taskrun/taskrun.go

```diff
@@ -184,6 +184,11 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, tr *v1beta1.TaskRun) pkg
 	// updates regardless of whether the reconciliation errored out.
 	if err = c.reconcile(ctx, tr, rtr); err != nil {
 		logger.Errorf("Reconcile: %v", err.Error())
+		if strings.Contains(err.Error(), "Result exceeded the maximum allowed limit") {
```

Comment:
I'm worried this may be too brittle - even in this PR I don't think this will end up matching because the error in status.go uses `result` instead of `Result` (we may want to add a test to check that it fails properly).

Can we base this off of some shared value - e.g. `errors.Is`, an exit code, or worst case from a shared error string variable?

Reply:
I based this off of errors.Is and a shared error string variable.